examples/server/server.cpp

```diff
@@ -453,23 +487,33 @@ int main(int argc, char ** argv) {
         std::vector<float> pcmf32;               // mono-channel F32 PCM
         std::vector<std::vector<float>> pcmf32s; // stereo-channel F32 PCM
 
-        // write to temporary file
-        const std::string temp_filename = "whisper_server_temp_file.wav";
-        std::ofstream temp_file{temp_filename, std::ios::binary};
+        // write file to temporary file
+        std::ofstream temp_file{filename, std::ios::binary};
```

Comment:
We can't create a file with the same name as the file we recieve, if the request/file is the in the same current working directory it will remove the file completely. Please keep the filename.

Reply:
> We can't create a file with the same name as the file we recieve, if the request/file is the in the same current working directory it will remove the file completely. Please keep the filename.

If I understand correctly, retaining the filename as specified by the user could pose a security risk.

Reply:
In a way it could, if you would name the file 'server' it would remove the binary and crash the server I think. I haven't tried but interesting take.